src/openeo_test_suite/tests/processes/processing/test_example.py

```diff
@@ -1,10 +1,177 @@
-def test_add35(connection):
+import json5
+import math
+from pathlib import Path, posixpath
+import pytest
+import xarray as xr
+import warnings
+
+# glob path to the test files
+examples_path = "assets/processes/tests/*.json5"
+
+def get_examples():
+    examples = []
+    package_root_folder = Path(__file__).parents[5]
+    files = package_root_folder.glob(examples_path)
+    for file in files:
+        id = file.stem
+        try:
+            with file.open() as f:
+                data = json5.load(f)
+                for test in data["tests"]:
+                    if "level" in test:
+                        level = test["level"]
+                    elif "level" in data:
+                        level = data["level"]
+                    else:
+                        level = "L4"
+                    
+                    examples.append([id, test, file, level])
+        except Exception as e:
+            warnings.warn("Failed to load {} due to {}".format(file, e))
+    
+    return examples
+
+@pytest.mark.parametrize("id,example,file,level", get_examples())
+def test_process(connection, process_levels, processes, id, example, file, level):
+    if len(process_levels) > 0 and level not in process_levels:
+        pytest.skip("Skipping process {} because {} is not in the specified levels: {}".format(id, level, ", ".join(process_levels)))
```

Comment:
In Python, so-called "f-strings" are usually a lot easier to use than these `"... {}...".format(..)` constructs:
```suggestion
        pytest.skip(f"Skipping process {id} because {level} is not in the specified levels: {process_levels}")
```

Reply:
Do you want me to change this? I had them all in f-strings, but then read that the format is the "new style" and such changed them all. 🙈

Reply:
No this is mostly informative, not merge-blocking

where id you read that? str.format is rather old-fashioned, f-strings is the modern way